### ISSUE TITLE
encoder.encodeMap() should not encodeNull() when map length is zero

### ISSUE THREAD

**Issue by WIZARDISHUNGRY** [aws/aws-sdk-go-v2#195]
### Version of AWS SDK for Go?

2.0.0-preview.4

### Version of Go (`go version`)?
`go version go1.10.3 darwin/amd64`

### What issue did you see?

My Dynamo records are being created with NULL entries instead of empty maps.

### Steps to reproduce
```go
type Message struct {
	ID            string `json:"id"`
	Likes   map[int]string `json:"likes"`

}
```
```go
// Create persist a message
func (d *MessageDao) Create(message *model.Message,) (*model.Message, error) {

	av, err := dynamodbattribute.MarshalMap(message)
	if err != nil {
		return nil, err
	}
	input := &dynamodb.PutItemInput{
		Item:      av,
		TableName: aws.String(d.tableName),
	}

	req := d.DynamoDB.PutItemRequest(input)
	_, err = req.Send()

	if err != nil {
		return nil, err
	}
	return message, nil
}
```
https://github.com/aws/aws-sdk-go-v2/blob/ac26486d921d336e20fec49e2afb348c801f0003/service/dynamodb/dynamodbattribute/encode.go#L335-L360

**Comment by WIZARDISHUNGRY** [comment]
see aws/aws-sdk-go#682

**Comment by xibz** [comment]
Hello @WIZARDISHUNGRY, thank you for reaching out to us. We've received this same issue in v1, please see [here](https://github.com/aws/aws-sdk-go/issues/682). The workaround will also work for v2. I am going to go ahead and mark this as a feature request. If you have any more questions or issues with workaround please let us know.

**Comment by WIZARDISHUNGRY** [comment]
For those arriving via search, etc. – a workaround for v2:
```go	
        av, err := dynamodbattribute.MarshalMap(message)
	emptyMap := make(map[string]dynamodb.AttributeValue)
	av["some map"] = dynamodb.AttributeValue{M: emptyMap}

**Comment by jasdel** [comment]
This is related to the v1 SDK work, aws/aws-sdk-go#2419

**Comment by jasdel** [comment]
This feature was implemented in #401 and released in v0.13.0

**Comment by mattopsh** [comment]
Hi @jasdel,
I see it has been closed, but could you provide an up to date example of how to use it in aws-sdk-go-v2? I cannot find any example in the docs.
